library/core/class.request.php

```diff
@@ -1655,6 +1681,7 @@ public function url($path = '', $withDomain = false, $ssl = null) {
             $result .= $hash;
         }
 
+        self::$urls[$staticKey] = ''.$result;
```

Comment:
Same kinda thing as the path: do we need to prefix the value with an empty string? If there's a need to cast a non-string value to a string, we could use a more obvious method (e.g. `strval` or [type casting](http://php.net/manual/en/language.types.type-juggling.php#language.types.typecasting)).

Reply:
done